nengo/utils/compat.py

```diff
@@ -12,6 +11,9 @@
 
 # If something's changed from Python 2 to 3, we handle that here
 if PY2:
+    warnings.warn("Python 2 compatibility has been dropped from Nengo as of "
+                  "version 3.0.0; nengo.utils.compat will be removed in the "
+                  "next minor release.")
```

Comment:
Do we have a process in place to ensure that things like this are in fact done before the release?

Reply:
Not really, we could add something to the [release guide](https://www.nengo.ai/releasing.html) to check for deprecated things and remove them if appropriate? Would that be enough process?

Reply:
Hmm it would be kind of awkward to get to the stage of doing the release and finding out that a PR needs to happen. I'm thinking more along the lines of triggering the TODO item to appear during the appropriate release cycle so that it's not forgotten about. I'll also make a note for myself to add this to `CHANGES.rst`.

Reply:
It's pretty common for things to come up in that first stage of the release, so I wouldn't worry about that too much. For example, we rarely run the slow tests, but we run them when releasing and that's required separate PRs a few times and it hasn't been a big issue.

Reply:
Resolved by additional commit to CHANGES.rst and this issue: https://github.com/nengo/nengo.github.io/issues/71